test/phoenix/transports/transport_test.exs

```diff
@@ -21,6 +25,7 @@ defmodule Phoenix.Transports.TransportTest do
 
   setup do
     Logger.disable(self())
+    on_exit fn -> Endpoint.config_change([{Endpoint, @config}], []) end
```

Comment:
Well, I found another bug when I did it like this.

In `force_ssl/4` there is also a lookup to the host of the url.